### PR TITLE
Improve Koa types

### PR DESCRIPTION
**Pull request by thewilkybarkid**
This should narrow down and clarify the expected context.

Refs https://github.com/libero/article-store/pull/115#discussion_r364136341.

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -19,6 +19,13 @@ module.exports = {
   ],
   root: true,
   rules: {
+    'max-len': ['error', 120, 2, {
+      ignoreUrls: true,
+      ignoreComments: false,
+      ignoreRegExpLiterals: true,
+      ignoreStrings: true,
+      ignoreTemplateLiterals: true,
+    }],
```
Comment (thewilkybarkid) [inline .eslintrc.js]: The Airbnb 100 limit is getting quite annoying. Will extract this change.

File: src/middleware/data-factory.ts [hunk 2]
```diff
@@ -1,14 +1,13 @@
...
 
-export type DataFactoryContext<Context extends DefaultContext = DefaultContext,
+export type DataFactoryContext<Context extends DefaultContextExtends = DefaultContextExtends,
   Factory extends DataFactory = DataFactory> = Context & { dataFactory: Factory };
 
-export default <Factory extends DataFactory>(dataFactory: Factory):
-  Middleware<DefaultState, DataFactoryContext<{}, Factory>> => (
-  async (context: DataFactoryContext<{}, Factory>, next: Next): Promise<void> => {
+export default (dataFactory: DataFactory): Middleware<DefaultStateExtends, DefaultContextExtends> => (
+  async (context: DefaultContextExtends, next: Next): Promise<void> => {
```
Comment (thewilkybarkid) [inline src/middleware/data-factory.ts:10]: This is like https://github.com/libero/article-store/pull/115#discussion_r363821502, the `dataFactory` context hasn't been set yet so using `DataFactoryContext` isn't right.

File: .eslintrc.js (hunk 1 above)
Comment (thewilkybarkid) [reply .eslintrc.js]: https://github.com/libero/article-store/pull/160

**Review by giorgiosironi — APPROVED** [review]
Consistent across the middlewares

**Pull request merged**